test/src/test/java/org/corfudb/runtime/collections/CorfuStoreSecondaryIndexTest.java

```diff
@@ -1064,13 +1064,13 @@ public void testInvalidNestedSecondaryIndexRoot() throws Exception {
         final String tableName = "ManagedMetadata";
 
         // Create & Register the table.
-        assertThrows(IllegalArgumentException.class, () -> shimStore.openTable(
+        assertThrows(UnsupportedOperationException.class, () -> shimStore.openTable(
```

Comment:
Why did this exception type change? Isn't it closer to an IllegalArgument as the secondary index is incorrectly defined, rather than an operation not supported.

Reply:
+1. I ran the test and it's throwing `UnsupportedOperationException ` because of 
```
Caused by: java.lang.UnsupportedOperationException: This field is not of message type.
	at com.google.protobuf.Descriptors$FieldDescriptor.getMessageType(Descriptors.java:1090)
	at org.corfudb.runtime.collections.ProtobufIndexer.validateNestedSecondaryKey(ProtobufIndexer.java:345)
	at org.corfudb.runtime.collections.ProtobufIndexer.registerNestedSecondaryIndex(ProtobufIndexer.java:303)
```
I think we should catch `UnsupportedOperationException` in `ProtobufIndexer.validateNestedSecondaryKey` and throw `IllegalArgumentException` instead.

Reply:
done